### ISSUE TITLE
`result and result_tree` ?

### ISSUE THREAD

**Issue by StevenClontz** [PreTeXtBook/pretext#1684]
https://github.com/PreTeXtBook/pretext/blob/5a8d61fc5e366bd02c4e0b8a8aaecdc7be2b5198/pretext/pretext.py#L2143

To prevent a nastier error after a previous error caught by [line 2128](https://github.com/PreTeXtBook/pretext/blob/5a8d61fc5e366bd02c4e0b8a8aaecdc7be2b5198/pretext/pretext.py#L2128), we could check here if `result_tree` was successfully defined and pass if not. Or maybe we put this within the higher try/except block?

**Comment by StevenClontz** [comment]
Right now I think any time the 2128 exception is invoked, we'll always get the following failure (via CLI):

```
critical: A fatal error has occurred:
critical:  local variable 'result_tree' referenced before assignment 
critical: For more info, run pretext with `-v debug`
```

**Comment by rbeezer** [comment]
Right - I see this one regularly.  So, spell it out for me.  Should we raise an (informative) error inside the 2128 `except` which would basically be the `error_log` with a bit more explanation?

**Comment by StevenClontz** [comment]
So the problem is we're handling an error in line 2128, then moving on despite the fact we know there will be an error in the next try/except block.

The solution is probably adding a bare `return` at the end of the except block in line 2128 to make sure we don't try to move on to certain doom, then replacing `os.chdir(owd)` with `finally: os.chdir(owd)` to make sure we change directories back even if that happens.

**Comment by rbeezer** [comment]
Wouldn't an error be better?  There's only worse doom coming if this fails here.

**Comment by rbeezer** [comment]
See #1685

**Comment by rbeezer** [comment]
Closed via #1685
